docs/source/typing_anti_pitch.rst

```diff
@@ -0,0 +1,80 @@
+
+Anway, with all that said, here's a list of possible reasons to not use static type checking
+in Python::
+
+* You simply don't want to. Python is a tool that is meant to serve you. Python is a big tent,
+  multi-paradigm language that generally allows you to do things in the way that best suits your
+  needs, as best determined by you.
+
+* The cost-benefit ratio isn't good enough. Pleasing static type checkers requires a non-zero amount
+  of busy work. If this isn't worth the extra confidence you get, you shouldn't add static type
+  checking.
+
+* Your codebase fits in your developers' heads. Opinions vary, but people tend to agree that at
+  some number of developers and lines of code, static type checking confers significantly more
+  benefit. You don't feel like you're there yet.
+
+* If you maintain high test coverage, that might provide sufficient quality assurance for your
+  needs (acknowledging that static type checking and tests enforce different things; static type
+  checking usually cannot validate logic, tests can often not prove invariants of your code to
+  hold).
+
+* Your codebase is old, large and has been working fine without static type checking for years.
+  While Python's type system is designed to
+  `allow gradual adoption of static type checking <https://mypy.readthedocs.io/en/stable/existing_code.html>`_,
+  the total cost of adding type annotations to a large extant codebase can be prohibitive.
+
+* Your library does enough dynamic things that type checking would be unlikely to help your users.
+  Either you would have to spend a lot of effort redesigning your library in ways that static type
+  checkers could better understand, or you'd have to invest a lot of effort into figuring out clever
+  type annotations to twist the arms of type checkers.
+
+* Your codebase has suffered at the hands of `Hyrum's Law <https://www.hyrumslaw.com/>`_
+  and all possible observable behaviour is depended on. In order to avoid false positives for your
+  users, all your types end up being either a) complicated ``Protocol``\s that are hard to maintain,
+  or b) ``Any`` in which case there's not much point. (On the other hand, static type checking could
+  be a good solution for communicating to users what behaviour they should be allowed to rely on)
+
+* You're not opposed to type checking in theory, but you dislike Python type checkers in practice.
+  Maybe they don't understand enough of the idioms you use, maybe you'd like them to infer more
+  instead of relying on explicit annotations, maybe they're too slow, maybe they don't integrate
+  well with your editor, maybe they're too hard to configure. Whatever the reason -- it just doesn't
+  work for your project.
+
+* Type checking in Python isn't actually strict enough, powerful enough or expressive enough for
+  you. Python type checkers end up making various decisions out of pragmatism, or due to limited
+  resources, and these decisions might not be the ones for you.
```

Comment:
While I find myself in this camp, I don't think this is a good point as an anti-pitch, if anything this pitcher to either 

a) use typing *in addition to* other checks
b) to not use Python for these tasks
or c) to create a type checker which is as strict as is desired.

Maybe there should be a section noting what type checking doesn't (currently) do? But then this is largely based on what implementations currently support. That's still somewhat Python's problem on a social level, having kicked this to external implementors and not having a reference implementation that others must conform with, but it's harder to have such a documentation section as a result as well.

either way though, I don't think this belongs, at least not without some modifications that point out that this isn't necessarily a "why not typing".

Reply:
Thanks, I added https://github.com/python/typing/pull/1477/commits/292145126fb053fa77eb0d77d33601ad05c3e41c
While this is a "why not typing", it is implicitly a "why not typing in Python as is implemented today", so I think it still belongs, even if it's unlike the other items in this list.